File: app/src/main/java/net/xzos/UpgradeAll/ui/viewmodels/log/PlaceholderFragment.java
package net.xzos.UpgradeAll.ui.viewmodels.log;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import net.xzos.UpgradeAll.R;
import net.xzos.UpgradeAll.ui.viewmodels.adapters.LogItemAdapter;

/**
 * A placeholder fragment containing a simple view.
 */
public class PlaceholderFragment extends Fragment {

    private static final String ARG_SECTION_NUMBER = "LogObjectTag";

    private Context mContext;

    private PageViewModel pageViewModel;

    static PlaceholderFragment newInstance(@NonNull String[] logObjectTag) {
        PlaceholderFragment fragment = new PlaceholderFragment();
        Bundle bundle = new Bundle();
        bundle.putStringArray(ARG_SECTION_NUMBER, logObjectTag);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mContext = getContext();
        pageViewModel = new ViewModelProvider(this).get(PageViewModel.class);
        String[] logObjectTag = {null, null};
        if (getArguments() != null) {
            logObjectTag = getArguments().getStringArray(ARG_SECTION_NUMBER);
        }
        pageViewModel.setLogObjectTag(logObjectTag);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_log, container, false);
        final RecyclerView logListView = root.findViewById(R.id.log_list);
        GridLayoutManager layoutManager = new GridLayoutManager(mContext, 1);
        logListView.setLayoutManager(layoutManager);
        LogItemAdapter adapter = new LogItemAdapter(pageViewModel.getLogList(), this);
        logListView.setAdapter(adapter);
        return root;
    }
}

File: app/src/main/kotlin/net/xzos/upgradeAll/ui/viewmodels/log/PlaceholderFragment.kt
package net.xzos.upgradeAll.ui.viewmodels.log

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView

import net.xzos.upgradeAll.R
import net.xzos.upgradeAll.ui.viewmodels.adapters.LogItemAdapter

/**
 * A placeholder fragment containing a simple view.
 */
class PlaceholderFragment : Fragment() {

    private lateinit var mContext: Context

    private lateinit var pageViewModel: PageViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = context!!
        pageViewModel = ViewModelProvider(this).get(PageViewModel::class.java)
        if (arguments != null) {
            val logObjectTag = arguments?.getStringArray(ARG_SECTION_NUMBER)
            if (logObjectTag != null)
                pageViewModel.setLogObjectTag(logObjectTag)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val root = inflater.inflate(R.layout.fragment_log, container, false)
        val logListView = root.findViewById<RecyclerView>(R.id.log_list)
        val layoutManager = GridLayoutManager(mContext, 1)
        logListView.layoutManager = layoutManager
        val adapter = LogItemAdapter(pageViewModel.logList, this)
        logListView.adapter = adapter
        return root
    }

    companion object {

        private const val ARG_SECTION_NUMBER = "LogObjectTag"

        internal fun newInstance(logObjectTag: Array<String>): PlaceholderFragment {
            val fragment = PlaceholderFragment()
            val bundle = Bundle()
            bundle.putStringArray(ARG_SECTION_NUMBER, logObjectTag)
            fragment.arguments = bundle
            return fragment
        }
    }
}